python/kfserving/kfserving/handlers/http.py

```diff
@@ -43,21 +47,46 @@ def validate(self, request):
             )
         return request
 
+    def has_ce_headers(self, headers: typing.Dict[str, str]) -> bool:
+        return (
+            "ce-specversion" in headers
+            and "ce-source" in headers
+            and "ce-type" in headers
+            and "ce-id" in headers
+        )
 
 class PredictHandler(HTTPHandler):
     async def post(self, name: str):
+        if self.has_ce_headers(self.request.headers):
+            body = from_http(self.request.headers, self.request.body)
```

Comment:
I think we also need to add catch exception for decoding error
from_http raises exceptions https://github.com/cloudevents/sdk-python/blob/master/cloudevents/http/http_methods.py#L14

Reply:
@wengyao04 is right, if we do not catch this then we tornado by default returns HTTP 500 for uncaught exception but here we really want HTTP 400(bad request).

Reply:
Good catch @wengyao04 .
I've updated this.